config/supported_assets.go

```diff
@@ -10,21 +10,22 @@ var (
 	// SupportedProviders defines a lookup table of all the supported currency API
 	// providers and whether or not they require an API key to be passed in.
 	SupportedProviders = map[provider.Name]APIKeyRequired{
-		provider.ProviderKraken:    false,
-		provider.ProviderBinance:   false,
-		provider.ProviderBinanceUS: false,
-		provider.ProviderCrescent:  false,
-		provider.ProviderOsmosis:   false,
-		provider.ProviderOsmosisV2: false,
-		provider.ProviderOkx:       false,
-		provider.ProviderHuobi:     false,
-		provider.ProviderGate:      false,
-		provider.ProviderCoinbase:  false,
-		provider.ProviderBitget:    false,
-		provider.ProviderMexc:      false,
-		provider.ProviderCrypto:    false,
-		provider.ProviderPolygon:   true,
-		provider.ProviderMock:      false,
+		provider.ProviderKraken:     false,
+		provider.ProviderBinance:    false,
+		provider.ProviderBinanceUS:  false,
+		provider.ProviderCrescent:   false,
+		provider.ProviderOsmosis:    false,
+		provider.ProviderOsmosisV2:  false,
+		provider.ProviderOkx:        false,
+		provider.ProviderHuobi:      false,
+		provider.ProviderGate:       false,
+		provider.ProviderCoinbase:   false,
+		provider.ProviderBitget:     false,
+		provider.ProviderMexc:       false,
+		provider.ProviderCrypto:     false,
+		provider.ProviderPolygon:    true,
+		provider.ProviderMock:       false,
+		provider.ProviderEthUniswap: false,
```

Comment:
Is there a reason to specify EthUniswap rather than just call it Uniswap provider?

Reply:
yeah, uniswap provider can support multiple uniswap providers, currently, this is for eth, but later it could have other evm chains as well, functioning as separate providers